runtime-transaction/src/instruction_details.rs

```diff
@@ -0,0 +1,199 @@
+use {
+    solana_builtins_default_costs::BUILTIN_INSTRUCTION_COSTS,
+    solana_compute_budget::compute_budget_limits::*,
+    solana_sdk::{
+        borsh1::try_from_slice_unchecked,
+        compute_budget::{self, ComputeBudgetInstruction},
+        instruction::{CompiledInstruction, InstructionError},
+        pubkey::Pubkey,
+        saturating_add_assign,
+        transaction::{Result, TransactionError},
+    },
+    std::num::NonZeroU32,
+};
+
+#[derive(Default, Debug)]
+struct ComputeBudgetInstructionDetails {
+    // compute-budget instruction details:
+    // the first field in tuple is instruction index, second field is the unsanitized value set by user
+    requested_compute_unit_limit: Option<(u8, u32)>,
```

Comment:
Uses to return `TransactionError::InstructionError` with index: https://github.com/anza-xyz/agave/blob/master/sdk/src/transaction/error.rs#L52-L55

compute-budget instructions are in static addresses range iirc, `u8` should be sufficient for now.

Reply:
Ah well,the `TransactionError::InstructionError` is actually wrong then 😆 

We can have more than 300 static/top-level instructions because there is no limit until that SIMD.

Even though transaction would fail we need to parse all to determine the fee...

This could give misleading results because the index would actually just be wrong, but since that SIMD is underway...maybe we just don't care. There's no practical reason for people to send such transaction right now since they cannot succeed

Reply:
> Why do we need the index at all though?

Following up on this. I see we use the index over the initial scan, but afaict we don't actually ever read it from the struct in sanitization. I think that still means we don't need to cache the index?

Reply:
sanitization is not currently reading it yet, cause in this PR, I intended to keep the behavior unchanged. I have few planned follow-ups to enhance sanitization here, such as using NonZeroU32 for cu-limits, and potentially #682, and perhaps others relate to pre-compiles/signatures; by then, sanitize could return `Err(TransactionError::InstructionErr(index, ...))`. I think it's no harm to keep instruction index.